src/socket_base.cpp

```diff
@@ -970,55 +979,56 @@ int zmq::socket_base_t::connect (const char *endpoint_uri_)
     //  Save last endpoint URI
     paddr->to_string (_last_endpoint);
 
-    add_endpoint (endpoint_uri_, static_cast<own_t *> (session), newpipe);
+    add_endpoint (make_unconnected_connect_endpoint_pair (_last_endpoint),
```

Comment:
@sigiesec this is what causes the TIPC issue - changing from endpoint_uri_ to _last_endpoint means that the user calling disconnect with endpoint_uri_ will not match anymore

Reply:
https://github.com/zeromq/libzmq/pull/3382

Reply:
Ah... I changed this somewhat by mistake. I think I noticed that in some contexts the original endpoint is used, but in some cases the resolved one. I think some uses need to be changed to provide meaningful monitor events, but apparently this is not consistent right now.

What I was thinking about: shouldn't it be possible to perform multiple wildcard binds on the same socket? Then each would bind to a different port/name. But then it doesn't make sense to store the original wildcard address for identifying an endpoint.

Reply:
yes it should be possible to do multiple wildcard binds - but this is in the connect call